src/installer/installer.go

```diff
@@ -634,14 +634,38 @@ func (i *installer) getInventoryHostsMap(hostsMap map[string]inventory_client.Ho
 
 func (i *installer) updateReadyMasters(nodes *v1.NodeList, readyMasters *[]string, inventoryHostsMap map[string]inventory_client.HostData) error {
 	nodeNameAndCondition := map[string][]v1.NodeCondition{}
+
+	// We are building a slice containing all the IP addresses of the nodes in the inventory so
+	// so that later we can match reporting nodes.
+	knownIpAddresses := []string{}
+	for _, v := range inventoryHostsMap {
+		knownIpAddresses = append(knownIpAddresses, v.IPs[:]...)
+	}
+
 	for _, node := range nodes.Items {
 		nodeNameAndCondition[node.Name] = node.Status.Conditions
 		if common.IsK8sNodeIsReady(node) && !funk.ContainsString(*readyMasters, node.Name) {
 			ctx := utils.GenerateRequestContext()
 			log := utils.RequestIDLogger(ctx, i.log)
 			log.Infof("Found a new ready master node %s with id %s", node.Name, node.Status.NodeInfo.SystemUUID)
 			*readyMasters = append(*readyMasters, node.Name)
+
+			// Matching of the node happens based on 2 rules
+			//   * if the name of the node and in the inventory is exactly the same, use use it
+			//   * if the name is not known in the inventory, we check if the IP address of the
+			//     reporting node is known to the inventory
+			// Using those rules we can cover the cases where e.g. inventory expects a short
+			// hostname, but the node reports itself using its FQDN
 			host, ok := inventoryHostsMap[strings.ToLower(node.Name)]
+			if !ok {
+				for _, ip := range node.Status.Addresses {
+					for _, knownIP := range knownIpAddresses {
```

Comment:
pfff... golang

Wonder if we want to over-engineer this and switch `knownIpAddress` to be a `map[IP]bool` just so we can do key checks.

Reply:
Yes of course 🎉